src/cow-react/common/hooks/useSafeMemo.ts

```diff
@@ -0,0 +1,25 @@
+import { useMemo } from 'react'
+import { CurrencyAmount, NativeCurrency, Token } from '@uniswap/sdk-core'
+
+/**
+ * UseMemo effectively (by values) compare only primitive types and compare objects by links
+ * To get the best performance we need process objects changes manually
+ */
+export function useSafeMemo<T>(memoCall: () => T, deps: unknown[] | null): T {
+  const safeDeps = deps
+    ? deps.map((dep) => {
```

Comment:
Maybe this function can be extracted and named. This way, we could reuse it for useCallback or useEffect

Reply:
Done